src/fromager/sources.py

```diff
@@ -281,6 +285,23 @@ def patch_source(ctx: context.WorkContext, source_root_dir: pathlib.Path) -> Non
             )
 
 
+def _warn_for_old_patch(
+    source_root_dir: pathlib.Path, path: str, version: Version
+) -> None:
+    if version:
+        # Get the existing version and req name as per the source_root_dir naming conventions
+        req_name = source_root_dir.name.rsplit("-", 1)[0]
+        existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
+        existing = req_name + "-" + str(existing_version)
+        new = req_name + "-" + str(version)
+
+        # Check to see if patches for older version exists
+        if version > existing_version and existing in path and new not in path:
```

Comment:
Using "in" with paths is problematic, because the substrings don't honor the path separators so you can end up matching because one name is a subset of another name.

The patch files are expected to be in a subdirectory that matches the source directory where the patch will be applied. I think that means you don't need to do so much parsing. Something like this should do it.

```python
if patch_filename.parent.name != source_root_dir.name:
    logger.warning("ignoring patch for ...")
```

Reply:
Okay, will make changes!